pkg/cmd/kind/delete/clusters/deleteclusters.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package clusters implements the `delete` command for multiple clusters
+package clusters
+
+import (
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/kind/pkg/errors"
+
+	"sigs.k8s.io/kind/pkg/cluster"
+	"sigs.k8s.io/kind/pkg/cmd"
+	"sigs.k8s.io/kind/pkg/log"
+)
+
+type flagpole struct {
+	Kubeconfig string
+	All        bool
+}
+
+// NewCommand returns a new cobra.Command for cluster creation
+func NewCommand(logger log.Logger, streams cmd.IOStreams) *cobra.Command {
+	flags := &flagpole{}
+	cmd := &cobra.Command{
+		Args: cobra.MinimumNArgs(0),
+		// TODO(bentheelder): more detailed usage
+		Use:   "clusters",
+		Short: "Deletes one or more clusters",
+		Long:  "Deletes a resource",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if !flags.All && len(args) == 0 {
+				return errors.New("no cluster names provided")
+			}
+
+			return deleteClusters(logger, flags, args)
+		},
+	}
+	cmd.Flags().StringVar(&flags.Kubeconfig, "kubeconfig", "", "sets kubeconfig path instead of $KUBECONFIG or $HOME/.kube/config")
+	cmd.Flags().BoolVar(&flags.All, "all", false, "delete all clusters")
+	return cmd
+}
+
+func deleteClusters(logger log.Logger, flags *flagpole, clusters []string) error {
+	provider := cluster.NewProvider(
+		cluster.ProviderWithLogger(logger),
+	)
+	if !flags.All {
```

Comment:
this path should just set ~`clusters := List()`

we can debug which set of clusters is deleted in the normal code path with a message like `Deleting clusters "kind-foo", "baz", "bar" ...`

Reply:
Yes, good call. I have changed this path to `if flags.All` and having the path set clusters to all of the clusters from `List()`. 

The `Deleting clusters` message lists clusters with the `[]` list frames around the cluster names, but it does accomplish listing all the clusters to be deleted.